shared/resourcemapping/src/main/java/com/google/cloud/opentelemetry/resource/ResourceTranslator.java

```diff
@@ -182,21 +182,25 @@ public static GcpResource mapResource(Resource resource) {
       case ResourceAttributes.CloudPlatformValues.GCP_APP_ENGINE:
         return mapBase(resource, "gae_instance", GOOGLE_CLOUD_APP_ENGINE_INSTANCE_LABELS);
       default:
-        // if k8s.cluster.name is set, pattern match for various k8s resources.
-        // this will also match non-cloud k8s platforms like minikube.
-        if (resource.getAttribute(ResourceAttributes.K8S_CLUSTER_NAME) != null) {
-          if (resource.getAttribute(ResourceAttributes.K8S_CONTAINER_NAME) != null) {
-            return mapBase(resource, "k8s_container", K8S_CONTAINER_LABELS);
-          } else if (resource.getAttribute(ResourceAttributes.K8S_POD_NAME) != null) {
-            return mapBase(resource, "k8s_pod", K8S_POD_LABELS);
-          } else if (resource.getAttribute(ResourceAttributes.K8S_NODE_NAME) != null) {
-            return mapBase(resource, "k8s_node", K8S_NODE_LABELS);
-          } else {
-            return mapBase(resource, "k8s_cluster", K8S_CLUSTER_LABELS);
-          }
-        }
-        return genericTaskOrNode(resource);
```

Comment:
If I'm reading this correctly, all of these lines are moved as-is to the `mapK8sResourceOrGenericTaskOrNode` function, correct ?

If this was indeed meant to be a refactor, I don't think we need it on account of my comment on Line 175.

Reply:
It is correct.